tests/Predis/Connection/FactoryTest.php

```diff
@@ -322,6 +322,39 @@ public function testCreateConnectionWithInitializationCommands(): void
         $prepareConnection->invoke($factory, $connection);
     }
 
+    /**
+     * @group disconnected
+     */
+    public function testCreateConnectionWithSkippedSetInfo(): void
+    {
+        $parameters = new Parameters([
+            'database' => '0',
+            'password' => 'foobar',
+            'set_client_info' => false,
+        ]);
+
+        $connection = $this->getMockBuilder('Predis\Connection\NodeConnectionInterface')->getMock();
+        $connection
+            ->expects($this->once())
+            ->method('getParameters')
+            ->willReturn($parameters);
+        $connection
+            ->expects($this->exactly(2))
+            ->method('addConnectCommand')
+            ->withConsecutive(
+                [$this->isRedisCommand('AUTH', ['foobar'])],
+                [$this->isRedisCommand('SELECT', ['0'])]
+            );
+
+        $factory = new Factory();
+
+        // TODO: using reflection to make a protected method accessible :facepalm:
+        $reflection = new ReflectionObject($factory);
```

Comment:
We can avoid using reflection object here, just call the public interface (e.g `$factory->create()`), `prepareConnection()` will be called as a part of execution script

Reply:
IMO there should be a refactor of the whole file for that for consistency. It's all over the place.

Reply:
Agreed, there's many things that can be refactored, but we still can avoid using Reflection, would you mind If I help you to fix all this issues in CI?

Reply:
UPD: Let's leave it for now, since all the tests are using this style of Reflection. Will be a part of another refactoring